examples/app-pytorch/driver.py

```diff
@@ -42,185 +47,110 @@ def weighted_average(metrics: List[Tuple[int, Metrics]]) -> Metrics:
-            req=pull_task_res_req
-        )
-        # ------------------------------------------------------------------ Driver SDK
-
-        task_res_list: List[task_pb2.TaskRes] = pull_task_res_res.task_res_list
-        print(f"Got {len(task_res_list)} results")
-
-        time.sleep(sleep_time)
-
-        all_task_res += task_res_list
-        if len(all_task_res) == len(task_ids):
-            break
-
-    # Collect correct results
-    node_messages: List[ClientMessage] = []
-    for task_res in all_task_res:
-        if task_res.task.HasField("legacy_client_message"):
-            node_messages.append(task_res.task.legacy_client_message)
-    print(f"Received {len(node_messages)} results")
-
-    weights_results: List[Tuple[NDArrays, int]] = []
-    metrics_results: List = []
-    for node_message in node_messages:
-        if not node_message.fit_res:
-            continue
-        fit_res = node_message.fit_res
-        # Aggregate only if the status is OK
-        if fit_res.status.code != Code.OK.value:
-            continue
-        weights_results.append(
-            (parameters_to_ndarrays(fit_res.parameters), fit_res.num_examples)
-        )
-        metrics_results.append(
-            (fit_res.num_examples, serde.metrics_from_proto(fit_res.metrics))
-        )
-
-    # Aggregate parameters (FedAvg)
-    parameters_aggregated = ndarrays_to_parameters(aggregate(weights_results))
-    parameters = parameters_aggregated
-
-    # Aggregate metrics
-    metrics_aggregated = weighted_average(metrics_results)
-    history.add_metrics_distributed_fit(
-        server_round=server_round, metrics=metrics_aggregated
-    )
-    print("Round ", server_round, " metrics: ", metrics_aggregated)
-
-    # Slow down the start of the next round
-    time.sleep(sleep_time)
-
-print("app_fit: losses_distributed %s", str(history.losses_distributed))
-print("app_fit: metrics_distributed_fit %s", str(history.metrics_distributed_fit))
-print("app_fit: metrics_distributed %s", str(history.metrics_distributed))
-print("app_fit: losses_centralized %s", str(history.losses_centralized))
-print("app_fit: metrics_centralized %s", str(history.metrics_centralized))
-
-# -------------------------------------------------------------------------- Driver SDK
-driver.disconnect()
-# -------------------------------------------------------------------------- Driver SDK
-print("Driver disconnected")
+        # Collect correct results
+        all_fitres = [
+            recordset_to_fitres(msg.content, keep_input=True) for msg in all_replies
+        ]
+        print(f"Received {len(all_fitres)} results")
+
+        weights_results: List[Tuple[NDArrays, int]] = []
+        metrics_results: List[Tuple[int, Dict]] = []
+        for fitres in all_fitres:
+            print(f"num_examples: {fitres.num_examples}, status: {fitres.status.code}")
+
+            # Aggregate only if the status is OK
+            if fitres.status.code != Code.OK.value:
```

Comment:
```suggestion
            if fitres.status.code != Code.OK:
```
`fitres.status.code` returns a `Code`, while `Code.OK.value` returned an int.